gemcalibration/include/gem/calib/Calibration.h

```diff
@@ -0,0 +1,224 @@
+                        {"trigType"  , "0"}, // TODO: TTC local should be only possible one
+                        {"l1aTime"   , "250"},
+                        {"pulseDelay", "40"},
+                        {"latency"   , "33"},
+                        {"armDacPoins","17,20,23,25,30,40,50,60,70,80,100,125,150,175"},
+                        // TODO: need to take the list of te ARM dac from a file
+                        }}
+            };
+
+            struct scanParamsRadioSelector{
+                std::string label;
+                std::vector<std::string>  options;
+            };
+
+            /**
+             *  map of the parameters for which radio selector are used with relative viable options
+             */
+
+            std::map<std::string, scanParamsRadioSelector> m_scanParamsRadioSelector{
+                {"trigType",{"TriggerType", {"TTC input","Loopback","Lemo/T3"}}},
+                {"signalSourceType", {"Signal Source", {"Calibration Pulse","Particle"}}},
+                {"comparatorType", {"Coparator Type", {"CDF","Arming comparator"}}},
+                {"adcType", {"VFAT ADC reference", {"Internal","External"}}},
+                {"perChannelType", {"DAC scan per channel", {"False","True"}}},
+            };
+            /**
+             *  set of parameters not filled with form
+             */
+            std::set <std::string> m_scanParamsNonForm  {"trigType", "signalSourceType", "signalSourceType", "comparatorType", "adcType", "perChannelType", "dacScanType" };
+
+            /**
+             *  map of the routine parameters and relative labels to appear in the interface
+             */
+
+            std::map< std::string,std::string > m_scanParamsLabels{
+                {"nSamples"  , "Number of samples"},
+                {"l1aTime"   , "L1A period (BX)"},
+                {"mspl"      , "Pulse stretch (int)"},
+                {"scanMin"   , "Scan min"},
+                {"scanMax"   , "Scan max"},
+                {"vfatChMin" , "VFAT Ch min"},
+                {"vfatChMax" , "VFAT Ch max"},
+                    //{"vt2"       , "CFG_THR_ARM_DAC"},// TODO:need to be taken from DB
+                {"trigThrottle"  , "Trigger throttle (int)"},
+                {"pulseDelay", "Pulse delay (BX)"},
+                {"latency"   , "Latency (BX)"},
+                {"trimValues", "Points in dac range"}, // TODO:need to be implemented properly in the back end in order to get a given number of points {-63,0,63}
+                {"phaseMin"  , "Phase min (int)"  },  
+                {"phaseMax"  , "Phase max (int)"  },
+                {"stepSize", "Step size (int)"},
+                {"armDacPoins", "ARM DAC points"}, 
+            };
+
+            std::map<std::string, uint32_t> m_amcOpticalLinks;
+
+            dacScanType_t m_dacScanType;
+
+            struct dacScanTypeFeature{
+                std::string label;
+                std::map<std::string, uint32_t> range;
+            };
```

Comment:
OK, let me be more explicit.
For the usage as I've understood it here, the type that would be most appropriate would be a `union`
```c++
union dacFeature {
   std::string label;
   uint16_t min;
   uint16_t max;
};
```

Reply:
As @cgalloni found, in fact `union` was *not* the appropriate type for this usage, but a `struct` with two additional fields rather than a `map` was.